app_backend/balance.py

```diff
@@ -0,0 +1,71 @@
+from user import User
+
+
+class BalanceTracker(User):
+    def __init__(self, uuid: str, dooley_dollars: float, meal_swipes: int):
+        super().__init__(uuid, dooley_dollars, meal_swipes)
+        """
+
+        name: str
+            The name of the user
+
+        dooley_dollars: float
+            The current balance of dooley dollars
+
+        meal_swipes: int
+            The current amount of remaining meal swipes
+
+        """
+
+    # ------------ Decrementing Balances (for spending) -------------
+
+    def dooley_sub(self, spent: float):
+        """
+        spent: The amount of dooley dollars spent
+        return: True or False (whether operation succeeds or fails)
+        """
+        if (spent < 0 ): # Error (negative value)
```

Comment:
This depends more on how you want to structure this method, but should users be allowed to subtract $0?